integration/bootstrap_before_buffer_rotation_no_tick_regression_test.go

```diff
@@ -0,0 +1,207 @@
+		SetNamespaces([]namespace.Metadata{ns1})
+
+	setup, err := newTestSetup(t, opts, nil)
+	require.NoError(t, err)
+	defer setup.close()
+
+	setup.mustSetTickMinimumInterval(100 * time.Millisecond)
+
+	// Setup the commitlog and write a single datapoint into it one second into the
+	// active block.
+	commitLogOpts := setup.storageOpts.CommitLogOptions().
+		SetFlushInterval(defaultIntegrationTestFlushInterval)
+	setup.storageOpts = setup.storageOpts.SetCommitLogOptions(commitLogOpts)
+
+	testID := ident.StringID("foo")
+	now := setup.getNowFn().Truncate(blockSize)
+	setup.setNowFn(now)
+	startTime := now
+	commitlogWrite := ts.Datapoint{
+		Timestamp: startTime.Add(time.Second),
+		Value:     1,
+	}
+	seriesMaps := map[xtime.UnixNano]generate.SeriesBlock{
+		xtime.ToUnixNano(startTime): generate.SeriesBlock{
+			generate.Series{
+				ID:   testID,
+				Data: []ts.Datapoint{commitlogWrite},
+			},
+		},
+	}
+	writeCommitLogData(t, setup, commitLogOpts, seriesMaps, ns1.ID())
+
+	// Setup bootstrappers - We order them such that the custom test bootstrapper runs first
+	// which does not bootstrap any data, but simply waits until it is signaled, allowing us
+	// to delay bootstrap completion until we've forced a tick to "hang". After the custom
+	// test bootstrapper completes, the commitlog bootstrapper will run.
+	bootstrapOpts := newDefaulTestResultOptions(setup.storageOpts)
+	bootstrapCommitlogOpts := bcl.NewOptions().
+		SetResultOptions(bootstrapOpts).
+		SetCommitLogOptions(commitLogOpts)
+	commitlogBootstrapper, err := bcl.NewCommitLogBootstrapper(bootstrapCommitlogOpts, nil)
+	require.NoError(t, err)
+
+	// Setup the test bootstrapper to only return success when a signal is sent.
+	signalCh := make(chan struct{})
+	test := newTestBootstrapperSource(testBootstrapperSourceOptions{
+		read: func(_ namespace.Metadata, shardTimeRanges result.ShardTimeRanges, _ bootstrap.RunOptions) (result.BootstrapResult, error) {
+			<-signalCh
+			result := result.NewBootstrapResult()
+			// Mark all as unfulfilled so the commitlog bootstrapper will be called after
+			result.SetUnfulfilled(shardTimeRanges)
+			return result, nil
+		},
+	}, bootstrapOpts, commitlogBootstrapper)
+
+	process := bootstrap.NewProcess(test, bootstrapOpts)
+	setup.storageOpts = setup.storageOpts.SetBootstrapProcess(process)
+
+	// Start a background goroutine which will wait until the server is started,
+	// issue a single write into the active block, change the time to be far enough
+	// into the *next* block that the series buffer for the previously active block
+	// can be rotated, and then signals the test bootstrapper that it can proceed.
+	go func() {
+		// Wait for server to start
+		setup.waitUntilServerIsUp()
+
+		// Set the time such that the (previously) active block is ready to be flushed.
+		now = now.Add(blockSize).Add(ropts.BufferPast()).Add(time.Second)
+		setup.setNowFn(now)
+
+		// Set the tick interval to be so large we can "hang" a tick at the end, preventing
```

Comment:
should the ordering of these be reversed?  i.e. first update the tick interval, and then update the `setup.NowFn`?

Reply:
I dont think it matters. Are you worried about a premature flush? It wont happen because the bootstrapper is blocked for the duration of this method

Reply:
Yeah fair. Disregard.